mesonbuild/backend/ninjabackend.py

```diff
@@ -2234,9 +2234,9 @@ def get_target_type_link_args(self, target, linker):
         if isinstance(target, build.Executable):
             # Currently only used with the Swift compiler to add '-emit-executable'
             commands += linker.get_std_exe_link_args()
-            # If gui_app, and that's significant on this platform
-            if target.gui_app and hasattr(linker, 'get_gui_app_args'):
-                commands += linker.get_gui_app_args()
+            # If gui_app is significant on this platform
+            if hasattr(linker, 'get_gui_app_args'):
+                commands += linker.get_gui_app_args(target.gui_app)
```

Comment:
I think we should define `get_gui_app_args()` to return `[]` in the base C compiler class so we don't have to do checks like this.

Reply:
Yeah, I think that makes sense.